Tests/SentryTests/Helper/SentryFileManagerTests.swift

```diff
@@ -663,55 +663,202 @@ class SentryFileManagerTests: XCTestCase {
+        let actualTracesSampleRate = try XCTUnwrap(config?[kSentryLaunchProfileConfigKeyTracesSampleRate] as? NSNumber).doubleValue
+        let actualProfilesSampleRate = try XCTUnwrap(config?[kSentryLaunchProfileConfigKeyProfilesSampleRate] as? NSNumber).doubleValue
+        expect(actualTracesSampleRate) == expectedTracesSampleRate
+        expect(actualProfilesSampleRate) == expectedProfilesSampleRate
+    }
+    
+    // if a file is still present in the primary location, like if a crash occurred before it could be removed, or an error occurred when trying to remove it or move it to the backup location, make sure we overwrite it
+    func testWriteAppLaunchProfilingConfigFile_fileAlreadyExists() throws {
+        try ensureAppLaunchProfileConfig(exists: true, tracesSampleRate: 0.75, profilesSampleRate: 0.75)
+        
+        let expectedTracesSampleRate = 0.12
+        let expectedProfilesSampleRate = 0.34
+        writeAppLaunchProfilingConfigFile([
+            kSentryLaunchProfileConfigKeyTracesSampleRate: expectedTracesSampleRate,
+            kSentryLaunchProfileConfigKeyProfilesSampleRate: expectedProfilesSampleRate
+        ])
+        
+        let config = NSDictionary(contentsOf: launchProfileConfigFileURL())
+        
+        let actualTracesSampleRate = try XCTUnwrap(config?[kSentryLaunchProfileConfigKeyTracesSampleRate] as? NSNumber).doubleValue
+        let actualProfilesSampleRate = try XCTUnwrap(config?[kSentryLaunchProfileConfigKeyProfilesSampleRate] as? NSNumber).doubleValue
+        expect(actualTracesSampleRate) == expectedTracesSampleRate
+        expect(actualProfilesSampleRate) == expectedProfilesSampleRate
+    }
+    
+    func testRemoveAppLaunchProfilingConfigFile() throws {
+        try ensureAppLaunchProfileConfig(exists: true)
+        expect(NSDictionary(contentsOf: launchProfileConfigFileURL())) != nil
+        removeAppLaunchProfilingConfigFile()
+        expect(NSDictionary(contentsOf: launchProfileConfigFileURL())) == nil
+    }
+    
+    // if there's not a file when we expect one, just make sure we don't crash
+    func testRemoveAppLaunchProfilingConfigFile_noFileExists() throws {
+        try ensureAppLaunchProfileConfig(exists: false)
+        expect(NSDictionary(contentsOf: launchProfileConfigFileURL())) == nil
+        removeAppLaunchProfilingConfigFile()
+        expect(NSDictionary(contentsOf: launchProfileConfigFileURL())) == nil
+    }
+    
+    func testBackupAppLaunchProfilingConfigFile() throws {
+        try ensureAppLaunchProfileConfig(exists: true)
+        try ensureAppLaunchProfileConfig(exists: false, backup: true)
+        expect(NSDictionary(contentsOf: launchProfileConfigFileURL())) != nil
+        expect(NSDictionary(contentsOf: launchProfileConfigBackupFileURL())) == nil
+        backupAppLaunchProfilingConfigFile()
+        expect(NSDictionary(contentsOf: launchProfileConfigFileURL())) == nil
+        expect(NSDictionary(contentsOf: launchProfileConfigBackupFileURL())) != nil
```

Comment:
`l`: I think validating the contents of the backup file similar to the test above would improve the quality of this test, as we don't check the contents anywhere else if I'm not mistaken.

https://github.com/getsentry/sentry-cocoa/blob/357449478f751f2664dd01495dcfddc07a31a939/Tests/SentryTests/Helper/SentryFileManagerTests.swift#L776-L781

Reply:
When I went back to read the codepaths, I realized that `appLaunchProfileConfiguration()` is only ever called from the spot where it's already reading from the backup location. We never actually read the sample rates at the time we check for whether to start the profiler, we only check for the existence of the file at that time. Here's the test that checks for the rates in the backup file: https://github.com/getsentry/sentry-cocoa/blob/357449478f751f2664dd01495dcfddc07a31a939/Tests/SentryTests/Helper/SentryFileManagerTests.swift#L730-L739

I'll make sure there's a high-level explanation of what's going on with this all. There are lots of headerdocs that explain each piece, like on the functions I've added to SentryFileManager and SentryLaunchProfiling, but nothing that really ties them together.